python/bind_data_access.h

```diff
@@ -261,11 +271,12 @@ template <class... Ts> class as_ElementArrayViewImpl {
       return py::cast(scalar, py::return_value_policy::move);
     } else {
       // Returning reference to element in variable. Return-policy
-      // reference_internal keeps alive `obj`. Note that an attempt to
+      // reference_internal keeps alive `parent`. Note that an attempt to
       // pass `keep_alive` as a call policy to `def_property` failed,
       // resulting in exception from pybind11, so we have handle it by
       // hand here.
-      return py::cast(scalar, py::return_value_policy::reference_internal, obj);
+      return py::cast(scalar, py::return_value_policy::reference_internal,
+                      std::move(parent));
```

Comment:
Does this affect only strings or does it fix the same issue for nested POD types?

Reply:
Do we have any POD types that can be stored in numpy arrays?
I have not tested with types that require ElementArrayView.

Reply:
I mean `double` etc?

Reply:
This was covered by #1824
But I will add an analogous test to the new one to make sure.

Reply:
Added a test with int64 and it passes.